test/planner/plan_util_test.cpp

```diff
@@ -163,7 +164,139 @@ TEST_F(PlanUtilTests, GetAffectedIndexesTest) {
+                                                 affected_cols_vector.end());
+
+  EXPECT_EQ(1, static_cast<int>(affected_cols.size()));
+  expected_oids.clear();
+  expected_oids.emplace(database_id, table_id, fname_col_oid);
+  EXPECT_EQ(expected_oids, affected_cols);
+
+  // ====== DELETE statements check ===
+  query_string = "DELETE FROM test_table;";
+  stmt.reset(new Statement("DELETE", query_string));
+  sql_stmt_list = peloton_parser.BuildParseTree(query_string);
+  affected_cols_vector = planner::PlanUtil::GetAffectedColumns(
+      txn->catalog_cache, std::move(sql_stmt_list), TEST_DB_COLUMNS);
+  affected_cols = std::set<planner::col_triplet>(affected_cols_vector.begin(),
+                                                 affected_cols_vector.end());
+
+  // all columns are affected
+  EXPECT_EQ(3, static_cast<int>(affected_cols.size()));
+  expected_oids.clear();
+  expected_oids.emplace(database_id, table_id, lname_col_oid);
+  expected_oids.emplace(database_id, table_id, id_col_oid);
+  expected_oids.emplace(database_id, table_id, fname_col_oid);
+  EXPECT_EQ(expected_oids, affected_cols);
+
+  // ========= INSERT statements check ==
+  query_string = "INSERT INTO test_table VALUES (1, 'pel', 'ton');";
+  stmt.reset(new Statement("INSERT", query_string));
+  sql_stmt_list = peloton_parser.BuildParseTree(query_string);
+  affected_cols_vector = planner::PlanUtil::GetAffectedColumns(
+      txn->catalog_cache, std::move(sql_stmt_list), TEST_DB_COLUMNS);
+  affected_cols = std::set<planner::col_triplet>(affected_cols_vector.begin(),
+                                                 affected_cols_vector.end());
+
+  // all columns are affected
+  EXPECT_EQ(3, static_cast<int>(affected_cols.size()));
+  expected_oids.clear();
+  expected_oids.emplace(database_id, table_id, lname_col_oid);
+  expected_oids.emplace(database_id, table_id, id_col_oid);
+  expected_oids.emplace(database_id, table_id, fname_col_oid);
+  EXPECT_EQ(expected_oids, affected_cols);
+
+  // ========= SELECT statement check ==
+  query_string = "SELECT id FROM test_table WHERE first_name = last_name;";
+  stmt.reset(new Statement("SELECT", query_string));
+  sql_stmt_list = peloton_parser.BuildParseTree(query_string);
+  affected_cols_vector = planner::PlanUtil::GetAffectedColumns(
+      txn->catalog_cache, std::move(sql_stmt_list), TEST_DB_COLUMNS);
+  affected_cols = std::set<planner::col_triplet>(affected_cols_vector.begin(),
+                                                 affected_cols_vector.end());
+
+  // all columns are affected
+  EXPECT_EQ(3, static_cast<int>(affected_cols.size()));
+  expected_oids.clear();
+  expected_oids.emplace(database_id, table_id, lname_col_oid);
+  expected_oids.emplace(database_id, table_id, id_col_oid);
+  expected_oids.emplace(database_id, table_id, fname_col_oid);
+  EXPECT_EQ(expected_oids, affected_cols);
+  txn_manager.CommitTransaction(txn);
 }
 
```

Comment:
I actually think the tests here are fine after you fix the above things. Eventually, we want to use the suggestion from the AutoAdmin algorithm so we don't need to overthink it for now.